docs/user-guide/data-structures/creating-arrays-and-datasets.ipynb

```diff
@@ -0,0 +1,616 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sc.full(dims=['x', 'y'], shape=[3, 4], value=1.23)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "Every filling function has a corresponding function with a `_like` postfix, for instance [scipp.zeros_like](../../generated/functions/scipp.zeros_like.rst).\n",
+    "These create a new variable (or data array) based on another variable (or data array):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "v = sc.array(dims=['x', 'y'], values=[[1, 2], [3, 4]], unit='m')\n",
+    "sc.zeros_like(v)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "### Special DTypes\n",
+    "\n",
+    "Scipp has a number of `dtypes` that require some conversion when creating variables.\n",
+    "Notably [scipp.datetimes](../../generated/functions/scipp.datetimes.rst), [scipp.vectors](../../generated/functions/scipp.vectors.rst), and their scalar counterparts [scipp.datetime](../../generated/functions/scipp.datetime.rst), [scipp.vector](../../generated/functions/scipp.vector.rst).\n",
+    "As well as types for spatial transformations in [scipp.spatial](../../generated/modules/scipp.spatial.rst).\n",
+    "While variables of all of these dtypes can be constructed using `scipp.array` and `scipp.scalar`, the specialized functions offer more convenience and document their intent better.\n",
+    "\n",
+    "`scipp.datetimes` constructs an array of date-time-points.\n",
+    "It can be called either with strings in ISO 8601 format:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sc.datetimes(dims=['time'], values=['2021-01-10T01:23:45',\n",
+    "                                    '2021-01-11T23:45:01'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Or with integers which encode the number of time units elapsed since the Unix epoc.\n",
+    "See also [scipp.epoch](../../generated/functions/scipp.epoch.rst)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sc.datetimes(dims=['time'], values=[0, 1610288175], unit='s')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "Note that the unit is mandatory in the second case.\n",
+    "\n",
+    "It is also possible to create a range of datetimes:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sc.arange('time', '2022-08-04T14:00:00', '2022-08-04T14:04:00',\n",
+    "          30, dtype='datetime64')"
```

Comment:
```suggestion
    "          num=30, dtype='datetime64')"
```
more readable?

Reply:
The fact that it should be `step` and not `num` shows that you are right :-)

Reply:
It is odd that we support a step without a unit here. If the datetime string was stored in a variable the code would yield unpredictable results?

Reply:
I think it is consistent. We require either all inputs are variables or none. The fact that the unit can be inferred from datetime strings is a bit special but we do it everywhere.

Reply:
A datetime string is conceptually equal to a scalar variable with a time unit. I think we should therefore also require a unit in the step size?

Reply:
Your argument only holds if the dtype is specified in addition because that overrides the 'natural dtype' of the string.
So in this case, `arange` gives special meaning to its arguments because it knows that it is constructing datetimes. But I would not add more special cases to it to make `step` to deviate from its usual requirements.

Reply:
Besides, this is not a discussion about this PR but about the implementation of `arange`. can we move it and finish up the PR?

Reply:
Sure, should move to issue.